server_test.go

```diff
@@ -180,64 +194,77 @@ type DDMetricsRequest struct {
 	Series []DDMetric
 }
 
+type channelMetricSink struct {
+	metricsChannel chan []samplers.InterMetric
+}
+
+func NewChannelMetricSink(ch chan []samplers.InterMetric) (*channelMetricSink, error) {
+	return &channelMetricSink{
+		metricsChannel: ch,
+	}, nil
+}
+
+func (c *channelMetricSink) Name() string {
+	return "channel"
+}
+
+func (c *channelMetricSink) Start(*trace.Client) error {
+	return nil
+}
+
+func (c *channelMetricSink) Flush(ctx context.Context, metrics []samplers.InterMetric) error {
+	// Put the whole slice in since many tests want to see all of them and we
+	// don't want them to have to loop over and wait on empty or something
+	c.metricsChannel <- metrics
+	return nil
+}
+
+func (c *channelMetricSink) FlushEventsChecks(ctx context.Context, events []samplers.UDPEvent, checks []samplers.UDPServiceCheck) {
+	return
+}
+
 // fixture sets up a mock Datadog API server and Veneur
 type fixture struct {
 	api             *httptest.Server
 	server          *Server
-	ddmetrics       chan DDMetricsRequest
 	interval        time.Duration
 	flushMaxPerBody int
 }
 
-func newFixture(t *testing.T, config Config) *fixture {
+func newFixture(t testing.TB, config Config, mSink metricSink, sSink spanSink) *fixture {
```

Comment:
It doesn't look like we're using the benchmark functions - what's the reason for changing to `TB`?

Reply:
It's use by `BenchmarkServerFlush`.